### PR TITLE
⚡ feat: add neovim config

### PR DESCRIPTION
**Pull request by melMass**
wip, I will edit this message better a bit later, but roughly:
image

image

- [x] propose a minimal neovim config à la [kickstart.nvim](https://github.com/nvim-lua/kickstart.nvim) (aka that you can copy paste) with a minimal setup to make nushell work a bit better with neovim.
  - [x] Syntax Highlighting using treesitter
  - [x] LSP using `nu --lsp`
  - [x] Basic `vim.opts` to play nicely with nushell

### CODE REVIEW

**Comment by melMass** [comment]
cc @amtoine if you can stress test it I'm sure you'll find issues :)

**Comment by fdncred** [comment]
I'm not seeing `nu --lsp` in the code. What am I missing?

**Comment by melMass** [comment]
The config is less then 150 lines of code so that it can be read...

**Comment by melMass** [comment]
Also you can literally test it in seconds if you follow the readme

**Comment by fdncred** [comment]
I can't get this to work the way I thought it would. The syntax highlighting and nushell lsp seem to work fine (congrats!), but I think it's loading cached files because even though I followed the readme, I still have my lunarvim colors. I wonder if there are some other env vars that it may be using or other folders?

My lunarvim setup does this
```nushell
$env.XDG_DATA_HOME = $env.APPDATA
$env.XDG_CONFIG_HOME = $env.LOCALAPPDATA
$env.XDG_CACHE_HOME = $env.TEMP

$env.LUNARVIM_RUNTIME_DIR = $'($env.XDG_DATA_HOME)\lunarvim'
$env.LUNARVIM_CONFIG_DIR = $'($env.XDG_CONFIG_HOME)\lvim'
$env.LUNARVIM_CACHE_DIR = $'($env.XDG_CACHE_HOME)\lvim'
$env.LUNARVIM_BASE_DIR = $'($env.LUNARVIM_RUNTIME_DIR)\lvim'
```

**Comment by melMass** [comment]
> I can't get this to work the way I thought it would.

If you use APPNAME it should use a separate dir for both the config file and the data so it's pretty strange, if you did not use APPNAME that's normal your data folder already has your old plugs and it create strange conflicts like this 

> Lastly, I still don't understand how it even knows about `nu --lsp` since you removed it from the init.lua. This is part of the reason why neovim is a pain for me.

Let me correct that to make it clearer it relies on too much magic currently.
I've since fully switched to neovim, something clicked

**Comment by melMass** [comment]
That should have created an isolated nvim-nu-data directory for you in LOCAPAPPDATA (or XDG on Unix)

**Comment by fdncred** [comment]
> That should have created an isolated nvim-nu-data directory for you in LOCAPAPPDATA (or XDG on Unix)

It did but there's also a C:\Temp\nvim-nu and a C:\Users\username\AppData\Roaming\nvim-nu-data and my regular lvim/lunarvim folders

**Comment by melMass** [comment]
I think this should instead be a markdown documentation of just the relevant bits. 
I will propose something later this week.

File: nvim/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+You should read it and copy/paste the relevant parts into your own config. 
+If you want to quickly test it in isolation you can use the [`$env.NVIM_APPNAME`](https://neovim.io/doc/user/starting.html#%24NVIM_APPNAME), here is how:
+
+## Testing the configuration in isolation
+
+- Copy or Symlink this folder to `$XDG_CONFIG_HOME/nvim-nu`:
+  - On unix it's usually: `$env.HOME/.config/nvim-nu`
+  - On Windows: `$env.LOCALAPPDATA/nvim-nu`
+
+- Run nvim using `NVIM_APPNAME=nvim-nu nvim`. This will run the config in isolation (with an isolated data folder).
```
Comment (amtoine) [inline nvim/README.md:14]: TIL `NVIM_APPNAME`, pretty cool, it even isolates `~/.local/share/nvim-nu/` :ok_hand:

File: nvim/init.lua [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+	-- WARN: this is optional
+	-- Highlight todo, notes, etc in comments
+	{ "folke/todo-comments.nvim", dependencies = { "nvim-lua/plenary.nvim" }, opts = { signs = false } },
+
+	-- NOTE: Use the official treesitter definition (nushell/tree-sitter-nu)
+	-- Syntax highlighing, code navigation etc..
+	-- The lua, vim and vimdoc one are optional but highly suggested for neovim.
+	{
+		"nvim-treesitter/nvim-treesitter",
```
Comment (amtoine) [inline nvim/init.lua:76]: afaik, there is an extra required step here to install the Nushell queries, right?

File: nvim/init.lua [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+end ---@diagnostic disable-next-line: undefined-field
+vim.opt.rtp:prepend(lazypath)
+
+-- INFO: settings to set nushell as the shell
+vim.opt.shellredir = "--stdin < %s"
+vim.opt.shellcmdflag = "-c"
+vim.opt.shellquote = ""
+vim.opt.shellxquote = " "
+vim.opt.shellxescape = ""
+vim.opt.sh = "nu"
```
Comment (amtoine) [inline nvim/init.lua:30]: still doubting this all works fine, but ehe

File: nvim/init.lua (hunk 2 above)
Comment (melMass) [reply nvim/init.lua:76]: I'm not using it and don't really understand the readme about it but in any case it shouldn't require an extra step imo (if it does 😅)

File: nvim/init.lua (hunk 3 above)
Comment (melMass) [reply nvim/init.lua:30]: I've been using it for quite some time, although I don't do very complex stuff with it, is solves most term calls used in nvim or by third party but I'd love to make it even better

File: nvim/init.lua (hunk 2 above)
Comment (amtoine) [reply nvim/init.lua:76]: aaaaah wait, `nu` is available in `treesitter` now???
that's so cool!!

**Comment by fdncred** [comment]
Let's go with this for now. We can always change it later if we don't like it. At least there is _much_ less magic going on now. Thanks @melMass!

**Comment by melMass** [comment]
Small note for windows users:

Because of how the os lock files, TSUpdate will fail if you have an opened buffer using the lang compiled ts definition.
For instance, updating TreeSitter-nu with a nushell file open will result in this:

image

**solution:** close nvim, reopen it without any file/session, TS should fire on its own, if not run `:TSUpdate`

**Comment by fdncred** [comment]
would be good to have that in the readme for this functionality.

**Pull request merged**